src/primitives/paraboloidsurface.jl

```diff
@@ -0,0 +1,115 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    ParaboloidSurface(point, radius, focallength)
+
+A paraboloid surface embedded in R³ and extending up to a distance `radius` from
+its focal axis, which is aligned along the z direction and passes through `point`
+(the apex of the paraboloid).
+
+    ParaboloidSurface(point, radius)
+
+This creates a paraboloid surface with focal length equal to 1.
+
+    ParaboloidSurface(point)
+
+This creates a paraboloid surface with focal length equal to 1 and a rim with unit
+radius.
+
+    ParaboloidSurface()
+
+Same as above, but here the vertex is at `Point(0, 0, 0)`.
+
+See also <https://en.wikipedia.org/wiki/Paraboloid>.
+"""
+struct ParaboloidSurface{T} <: Primitive{3, T}
+    point::Point{3,T}
+    radius::T
+    focallength::T
+end
+
+function ParaboloidSurface(point::Tuple, radius, focallength)
+    pt = Point(point)
+    ParaboloidSurface(pt, convert(eltype(pt), radius), convert(eltype(pt), focallength))
+end
+
+ParaboloidSurface(point::Point{3,T}, radius) where {T} = ParaboloidSurface(point, T(radius), one(T))
+
+ParaboloidSurface(point::Point{3,T}) where {T} = ParaboloidSurface(point, T(1), T(1))
+
+ParaboloidSurface{T}() where {T} = ParaboloidSurface(Point{3, T}(0, 0, 0), T(1), T(1))
+
+ParaboloidSurface() = ParaboloidSurface{Float64}()
+
+@doc raw"""
+    point(p::ParaboloidSurface{T})
+
+Return the vertex of the paraboloid as a `Point{3, T}`.
+"""
+point(p::ParaboloidSurface) = p.point
+
+@doc raw"""
```

Comment:
try to follow the docstring convention of other primitives, we don't use `@doc raw` at the moment.

Reply:
Ok, I am fixing this. I always use `@doc raw` because it's nicer if you use LaTeX formulae (as I did in this PR). For the moment, I just doubled all the occurrences of `\`.